test/provider_test.dart

```diff
@@ -113,6 +109,14 @@ void main() {
       expect(buildCount, equals(2));
     });
 
+    testWidgets('throws an error if no provider found', (tester) async {
+      await tester.pumpWidget(
+        const Builder(builder: Provider.of),
+      );
+
+      expect(tester.takeException(), isInstanceOf<ProviderNotFoundError>());
```

Comment:
I think `isInstanceOf` is deprecated

Since we're doing custom stuff with the error message, I think we should test the message too. 

```dart
expect(..., const TypeMatcher<ProviderNotFoundError>()
 .having((err) => err.message, "message", <contains the widget name>);
```

Reply:
Ermagerd, the Test APIs are so hard to use right now as they're transitioning to this new setup :/

Flutter has its own implementation of `TypeMatcher` in the Widgets package, so when I was attempting to use that it wasn't working (because it's not a test matcher).

In addition, `isInstanceOf` isn't deprecated in `flutter_test`, which is why we didn't see any analysis errors on Travis.

That said, it looks like I can do some annoying imports tricks to hide the `TypeMatcher` from the widget framework and use the `TypeMatcher` directly from the matchers package, which gives ya the additional `having `power.

Reply:
Changed the tests. I've implemented a simple `==` function on the Error, which means we don't have to do any of that type matching, we can just say `throwsA(ProviderNotFoundError('valueType', 'widgetType'))`

Reply:
I'm not a fan of using == for this, but it's fine. Thanks :smile: